### ISSUE TITLE
Replace Autocomplete library

### ISSUE THREAD

**Issue by parttimenerd** [leih-lokal/LeihLokalVerwaltung#204]
Replace the library used for autocompletions, as it currently doesn't support:
- validation
- required attribute

We can either replace it fully or create a fork. And we could think about adding all streets of Karlsruhe (https://github.com/CodeforKarlsruhe/streetnames) and other information to the autocompletions. A basic validation could also be useful (e.g. for e-mail addresses).

**Comment by daniel17903** [comment]
If there is another library supporting all required features we can replace it. If not I suggest that we implement these two features by referencing the input element of the current AutocompleteInput directly. Because imho it's not worth it to write or maintain a lot of additional code ourselves just for these two features.

For the street names I really like our current solution which uses the streets that already exist in our database for autocompletion.

Basic validation for mail addresses would be great.

**Comment by parttimenerd** [comment]
> For the street names I really like our current solution which uses the streets that already exist in our database for autocompletion.
May I ask why?

**Comment by skjerns** [comment]
> > For the street names I really like our current solution which uses the streets that already exist in our database for autocompletion.
> 
> May I ask why?

This makes it more flexible when porting it to other cities. Alternative would be to include a list of all streets of a pre-selected city. And in my experience there's a 80% chance that your street is already in

**Comment by parttimenerd** [comment]
We could also use a resource file, that is only loaded for this particular form? It would decrease the possibility of typos.

**Comment by skjerns** [comment]
Yes sure. We just need to make sure it's easily portable to other cities without people having to adjust anything manually within files.

**Comment by daniel17903** [comment]
> > For the street names I really like our current solution which uses the streets that already exist in our database for autocompletion.
> 
> May I ask why?

- As skjerns said it's portable without any effort
- We do not need to maintain and load a large list of streets
- The loading and searching of a large list of streets in the frontend might be slow. When using a resource file we would have to search it in the frontend. Only if we store the streets in the database we could search in the backend as we do with the current solution.
- It might be an overkill to load all streets of Karlsruhe because most customers live in a smaller area around the leihlokal.

To see how well the current solution works it might be interesting to generate a statistic about how often a user wants to enter a street which is not in the database yet.

**Comment by parttimenerd** [comment]
Thanks for the clarification. I keep this issue open until I or someone else implemented the validation and required attribute, as both should increase the usability.
